code/go/0chain.net/smartcontract/dbs/event/user.go

```diff
@@ -48,6 +54,111 @@ func (edb *EventDb) addOrUpdateUsers(users []User) error {
 	}).Create(&users).Error
 }
 
+func (edb *EventDb) updateUserCollectedRewards(users []User) error {
+	var ids []string
+	var collectedRewards []uint64
+	for _, u := range users {
+		ids = append(ids, u.UserID)
+		collectedRewards = append(collectedRewards, uint64(u.CollectedReward))
+	}
+
+	return CreateBuilder("users", "user_id", ids).
+		AddUpdate("collected_reward", collectedRewards, "users.collected_reward + t.collected_reward").Exec(edb).Error
+}
+
+func (edb *EventDb) updateUserTotalStake(users []User, shouldIncrease bool) error {
+	var ids []string
+	var stakes []uint64
+	for _, u := range users {
+		ids = append(ids, u.UserID)
+		stakes = append(stakes, uint64(u.TotalStake))
+	}
+
+	operation := "+"
+	if !shouldIncrease {
+		operation = "-"
+	}
+
+	return CreateBuilder("users", "user_id", ids).
+		AddUpdate("total_stake", stakes, "users.total_stake "+operation+" t.total_stake").Exec(edb).Error
+}
+
+func (edb *EventDb) updateUserReadPoolTotal(users []User, shouldIncrease bool) error {
+	var ids []string
+	var readpools []uint64
+	for _, u := range users {
+		ids = append(ids, u.UserID)
+		readpools = append(readpools, uint64(u.ReadPoolTotal))
+	}
+	operation := "+"
+	if !shouldIncrease {
+		operation = "-"
+	}
+
+	return CreateBuilder("users", "user_id", ids).
+		AddUpdate("read_pool_total", readpools, "users.read_pool_total "+operation+" t.read_pool_total").Exec(edb).Error
+}
+
+func (edb *EventDb) updateUserWritePoolTotal(users []User, shouldIncrease bool) error {
+	var ids []string
+	var writepools []uint64
+	for _, u := range users {
+		ids = append(ids, u.UserID)
+		writepools = append(writepools, uint64(u.WritePoolTotal))
+	}
+
+	operation := "+"
```

Comment:
we do not this hack, simply convert `u.WritePoolTotal` to int64 and use negative if you need to decrease